sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java

```diff
@@ -550,6 +552,12 @@ public void encode(
     }
 
     @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+      Coder<?> elementCoder = listCoder.getCoderArguments().get(0);
```

Comment:
Notice that you could just keep `elementCoder` in a field (as well as caching `listCoder`) on line 536. No need to pull it out here.